packages/eslint-plugin/tests/rules/require-closing-tags.test.js

```diff
@@ -165,5 +201,50 @@ ruleTester.run("require-closing-tags", rule, {
         },
       ],
     },
+    {
+      code: `<custom-tag id="foo" />`,
+      options: [
+        {
+          selfClosing: "always",
+          allowSelfClosingCustom: true,
+          customPatterns: [":"],
+        },
+      ],
+      output: null,
+      errors: [
+        {
+          messageId: "unexpected",
+        },
+      ],
+    },
+    {
+      code: `<custom-tag></custom-tag>`,
+      options: [
+        {
+          selfClosing: "always",
+          allowSelfClosingCustom: true,
+        },
+      ],
+      output: "<custom-tag /></custom-tag>",
```

Comment:
the rule fixes the opening tag to be self-closing, but it doesn't change the other end; i believe this is a known issue described in #171 (it looks a little strange here, but since the component could contain children it probably makes sense to leave the rest of the code unchanged and let the user clean that up themselves?)

Reply:
This is somewhat different from issue #171. The autofixed output should be correct HTML. If it changes to self closing, we need to remove the closing tag. I think we could use `fixer.remove(nodeOrToken)`.